SentryTestUtils/TestSentryDispatchQueueWrapper.swift

```diff
@@ -13,8 +16,12 @@ public class TestSentryDispatchQueueWrapper: SentryDispatchQueueWrapper {
     public var dispatchAsyncInvocations = Invocations<() -> Void>()
     public var dispatchAsyncExecutesBlock = true
     public override func dispatchAsync(_ block: @escaping () -> Void) {
-        dispatchAsyncCalled += 1
-        dispatchAsyncInvocations.record(block)
+        
+        dispatchAsyncLock.synchronized {
```

Comment:
This was a data race happening frequently when running the tests and it was a small easy fix. That's why I added it in this PR.